esmvaltool/_recipe.py

```diff
@@ -399,6 +399,19 @@ def _update_fx_settings(settings, variable, config_user):
             settings['mask_landseaice']['fx_files'].append(
                 fx_files_dict['sftgif'])
 
+    fx_keys = set(['average_region', 'area_average', 'volume_average',
```

Comment:
I think it would be better to do something like this here:
```python
for step in ('average_region', 'average_volume'):
    if step in settings and settings[step]['fx_files'] is not False:
        var = dict(variable)
        var['fx_files'] = ['some_required_type_of_fx_file', 'some_other_required_type_of_fx_file']
        settings[step]['fx_files'] = get_input_fx_filelist(
             variable=var,
             rootpath=config_user['rootpath'],
             drs=config_user['drs'],
        )
```
and also replace `variable = dict(variable)` by `var = dict(variable)` in the code above, to make sure the other preprocessor functions do not mess up the `variable` dict you use.

Reply:
Note that this will allow disabling the use of fx_files by settings `fx_files: false` in the recipe.

Reply:
As I mentionned above, I've had a go at removing the use_fx_files from _area_pp.py and the BGC recipe (as reqested by @bouweandela), but it had some strange effects and unusual reactions, like it caused some tests to fail. I think this may well have to wait until a future patch.

Reply:
Can you be more specific? I just tried removing it and the unit tests work just fine, this appears to be untested.

Reply:
I've made the changes to remove `use_fx_files` in this branch:  https://github.com/ESMValGroup/ESMValTool/tree/version2_development_remove_usefxfiles

I don't fully understand the implications of line 405 of [_recipe.py](https://github.com/ESMValGroup/ESMValTool/blob/version2_development_remove_usefxfiles/esmvaltool/_recipe.py#L405). 
```
var['fx_files'] = [] 
```
If we leave this dictionary entry as a empty list, like this, will this replace the fx_files loaded from the recipe?

Is this the right way to do this? I don't fully understand how this part works and I hesitate to make any changes.

Reply:
yeah that line is not correct; `var['fx_files']` is the list of keys that the fx file getter will populate with fullpaths to fx files keyed by the name of the fx variable eg `var = {'files': {'dataset1': 'path1'}, 'fx_files': {'sftlf': 'path_to_sftlf_file'}}` and by setting it to an empty list you are not retrieving you fx files correctly

Reply:
why not uncomment the commented bit and set `var['fx_files'] = ['areacello', 'volcello']` or add more fx variables as per the need of the two averaging functions? two

Reply:
@valeriupredoi, when I set this line to 
```
var['fx_files'] = ['areacello', 'volcello']
```
I got the error:
```
  File "/users/modellers/ledm/workspace/ESMValToolTest/ESMValTool/esmvaltool/preprocessor/_area_pp.py", line 184, in area_average
    '({})'.format(cube.data.shape, grid_areas.shape))
ValueError: Cube shape ((48, 216, 360)) doesn`t match grid area shape ((40, 216, 360))
```

However, the error is gone when I replace this with the lines:
```
            if step == 'average_region':
                var['fx_files'] = ['areacello', ]
            if step == 'average_volume':
                var['fx_files'] = ['volcello', ]
```
This may be a problem if someone wants to use a different area/volume field (like `areacella`/`volvella`, but I don't know a situation where that would be necessary.

Reply:
yes because this way you are applying a volume mask to a an area file, the separation you are doing by area or volume is probably the right way; in any case, what was wrong with setting the needed fx variables in the preprocessor module settings?

Reply:
> This may be a problem if someone wants to use a different area/volume field (like areacella/volvella, but I don't know a situation where that would be necessary.

You could make this configurable in the recipe, i.e. for average_region use areacella by default, or whatever is specified in fx_files  in the recipe and similarly for average_volume.

Reply:
I feel like this is becoming increasingly outside the scope of the PR (we're not even using the same branch anymore). Shall I create a new PR so we can merge that one in?